plugins/python-build/README.md

```diff
@@ -175,6 +175,15 @@ common for performance improvement from this to be in the ballpark of 30%.
 env PYTHON_CONFIGURE_OPTS='--enable-optimizations --with-lto' PYTHON_CFLAGS='-march=native -mtune=native' pyenv install 3.6.0
 ```
 
+If you are using a Mac computer with Apple silicon (e.g. M1), you need to install
+`clang` 15+ before running the command above and choose a recent minor version
+of Python. If you use Homebrew to install `clang` (`brew install llvm`), override
+the compiler to use:
+
+```sh
+env PATH="/opt/homebrew/opt/llvm/bin:$PATH" PYTHON_CONFIGURE_OPTS='--enable-optimizations --with-lto' PYTHON_CFLAGS='-march=native -mtune=native' pyenv install 3.10
```

Comment:
Did you test this command? [We have many reports of the build breaking when Homebrew's LLVM is on PATH](https://www.google.com/search?q=Homebrew+LLVM+on+PATH+site%253Agithub.com%252Fpyenv%252Fpyenv):

Reply:
I did, and seemed to have worked correctly. I installed by 3.10 (3.10.10) and 3.11 (3.11.2) using the exact command above. I'm using macOS 12.5.1 on an M1.

Reply:
Besides, is the result compatible with PyPI-hosted wheels? E.g. GCC-built Python has a different architecture tag.
Apple has been using `arm` instead of the standard `arm64` the last time I checked.

Reply:
If it didn't build them from source, the architectures must be compatible.

This is not an officially supported setup though so if it goes in, it must be with a warning that this has not been extensively tested and might break of have subtle incompatibilities, use at your own risk...

I don't think the README should be advertising unsupported setups. The wiki would be fine.

Plus, this is only temporarily relevant, until Apple ships CLang 15.

Reply:
I didn't build from source. Please feel free to move this information to the wiki or add the necessary disclaimers. I spent quite some time solving it so I figured I'd share the solution for others.

Reply:
I didn't build from source. Please feel free to move this information to the wiki or add the necessary disclaimers. I spent quite some time solving it so I figured I'd share the solution for others. :)

Reply:
Here: https://github.com/pyenv/pyenv/wiki/Build-for-maximum-performance-in-MacOS

The wiki has been set up to be user-editable so you can tune that text to your heart's content.

Reply:
Looks great, please feel free to close this PR.